extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java

```diff
@@ -394,6 +433,17 @@ private void handleNewWorkOrder(
     kernelHolder.partitionBoundariesFuture = runWorkOrder.getStagePartitionBoundariesFuture();
   }
 
+  @SuppressWarnings("unchecked")
+  private void updateMetricDimensions(QueryDefinition queryDef)
+  {
+    Set<String> datasourceDim = (Set<String>) queryMetricDimensions.computeIfAbsent(DruidMetrics.DATASOURCE, s -> new HashSet<>());
```

Comment:
note: I think it's true that all `workOrder`-s will be for the same `queryDef` - is that correct ?

Reply:
Currently, that is true, yes